testsuite/integration/basic/src/test/java/org/wildfly/test/integration/jmx/rbac/AbstractJmxAccessFromDeploymentWithRbacTest.java

```diff
@@ -0,0 +1,256 @@
+                APPLICATION_ROLES_PROPERTIES,
+                APPLICATION_USERS_PROPERTIES,
+                MGMT_GROUPS_PROPERTIES,
+                MGMT_USERS_PROPERTIES};
+
+        private final boolean useIdentityRoles;
+        private final Set<String> skippedFiles = new HashSet<>();
+
+        EnableRbacSetupTask(boolean useIdentityRoles) {
+            this.useIdentityRoles = useIdentityRoles;
+            if (!useIdentityRoles) {
+                skippedFiles.add(MGMT_GROUPS_PROPERTIES);
+            }
+        }
+
+        @Override
+        protected void doSetup(ManagementClient managementClient, String containerId) throws Exception {
+            super.doSetup(managementClient, containerId);
+            // Essentially we overwrite all the properties files for the Application- and ManagementRealm/Domains.
+            // We restore them in nonManagementCleanUp()
+            backupOrRestoreConfigDirPropertyFiles(true);
+            copyPropertiesToConfigFolder();
+
+
+            List<ModelNode> operations = new ArrayList<>();
+
+            // /core-service=management/access=authorization:write-attribute(name=provider,value=rbac)
+            PathAddress addr = PathAddress.pathAddress(CORE_SERVICE, MANAGEMENT).append(ACCESS, AUTHORIZATION);
+            operations.add(Util.getWriteAttributeOperation(addr, PROVIDER, "rbac"));
+
+            if (useIdentityRoles) {
+                // /core-service=management/access=authorization:write-attribute(name=use-identity-roles,value=true)
+                addr = PathAddress.pathAddress(CORE_SERVICE, MANAGEMENT).append(ACCESS, AUTHORIZATION);
+                operations.add(Util.getWriteAttributeOperation(addr, USE_IDENTITY_ROLES, true));
+            } else {
+                // /core-service=management/access=authorization/role-mapping=SuperUser/include=user-kabir:add(name=kabir,type=USER)
+                addr = PathAddress.pathAddress(CORE_SERVICE, MANAGEMENT)
+                        .append(ACCESS, AUTHORIZATION)
+                        .append(ROLE_MAPPING, "SuperUser")
+                        .append(INCLUDE, "user-kabir");
+                ModelNode addSuperUserInclude = Util.createAddOperation(addr);
+                addSuperUserInclude.get(NAME, "kabir");
+                addSuperUserInclude.get(TYPE, "USER");
+                operations.add(addSuperUserInclude);
+            }
+
+            // /core-service=management/access=identity:add(security-domain=ManagementDomain)
+            addr = PathAddress.pathAddress(CORE_SERVICE, MANAGEMENT).append(ACCESS, IDENTITY);
+            ModelNode addAccessIdentity = Util.createAddOperation(addr);
+            addAccessIdentity.get("security-domain").set("ManagementDomain");
+            operations.add(addAccessIdentity);
+
+            // /subsystem=undertow/application-security-domain=other:write-attribute(name=http-authentication-factory,value=application-http-authentication)
```

Comment:
I don't think this is necessary, the default `application-security-domain` configuration should work here.